File: screenShotServer/sssEclipseProject/src/adshotrunner/system/ASRProperties.java
package adshotrunner.system;

import java.io.FileInputStream;
import java.util.Properties;

public class ASRProperties {

	//---------------------------------------------------------------------------------------
	//---------------------------------- Constants ------------------------------------------
	//---------------------------------------------------------------------------------------	
	final private static String PROPERTIESFILE = "config/asr.properties";

	//---------------------------------------------------------------------------------------
	//-------------------------------- Static Variables -------------------------------------
	//---------------------------------------------------------------------------------------	
	//***************************** Public Static Variables *********************************

	//**************************** Protected Static Variables *******************************


	//***************************** Private Static Variables ********************************
	private static Properties asrProperties = null;

	//---------------------------------------------------------------------------------------
	//--------------------------------- Static Methods --------------------------------------
	//---------------------------------------------------------------------------------------	
	//***************************** Public Static Methods ***********************************
	//Database properties
	public static String asrDatabaseHost() {return getProperties().getProperty("asrDatabaseHost");}
	public static String asrDatabaseUsername() {return getProperties().getProperty("asrDatabaseUsername");}
	public static String asrDatabasePassword() {return getProperties().getProperty("asrDatabasePassword");}
	public static String asrDatabase() {return getProperties().getProperty("asrDatabase");}

	//ASR Domain
	public static String asrDomain() {return getProperties().getProperty("asrDomain");}

	//AWS Credentials
	public static String awsAccessKey() {return getProperties().getProperty("awsAccessKey");}
	public static String awsSecretKey() {return getProperties().getProperty("awsSecretKey");}
	public static String awsRegion() {return getProperties().getProperty("awsRegion");}

	//File storage containers
	public static String containerForCampaignJobs() {return getProperties().getProperty("containerForCampaignJobs");}
	public static String containerForPowerPoints() {return getProperties().getProperty("containerForPowerPoints");}
	public static String containerForPowerPointBackgrounds() {return getProperties().getProperty("containerForPowerPointBackgrounds");}
	public static String containerForScreenshots() {return getProperties().getProperty("containerForScreenshots");}
	public static String containerForTagImages() {return getProperties().getProperty("containerForTagImages");}
	public static String containerForTagPages() {return getProperties().getProperty("containerForTagPages");}
	public static String containerForTagTexts() {return getProperties().getProperty("containerForTagTexts");}

	//Email Addresses
	public static String emailAddressScreenshots() {return getProperties().getProperty("emailAddressScreenshots");}
	
	//Notification groups
	public static String notificationGroupForFrontend() {return getProperties().getProperty("notificationGroupForFrontend");}
	public static String notificationGroupForSSS() {return getProperties().getProperty("notificationGroupForSSS");}
	
	//Paths
	public static String pathForAdInjecterJavascript() {return getProperties().getProperty("pathForAdInjecterJavascript");}
	public static String pathForAdMarkerExtension() {return getProperties().getProperty("pathForAdMarkerExtension");}
	public static String pathForDefaultBackground() {return getProperties().getProperty("pathForDefaultBackground");}
	public static String pathForHTMLWithoutAnchorsJavascript() {return getProperties().getProperty("pathForHTMLWithoutAnchorsJavascript");}
	public static String pathForPhantomJS() {return getProperties().getProperty("pathForPhantomJS");}
	public static String pathForPossibleStoriesJavascript() {return getProperties().getProperty("pathForPossibleStoriesJavascript");}
	public static String pathForTemporaryFiles() {return getProperties().getProperty("pathForTemporaryFiles");}
	
	//Message queues
	public static String queueForScreenshotRequests() {return getProperties().getProperty("queueForScreenshotRequests");}
	public static String queueForTagImageRequests() {return getProperties().getProperty("queueForTagImageRequests");}
	
	//Selenium Hub information
	public static String seleniumHubURL() {return getProperties().getProperty("seleniumHubURL");}
	
	
	//**************************** Protected Static Methods *********************************


	//***************************** Private Static Methods **********************************
	private static Properties getProperties() {
		
		//Get the properties from the file if not yet done
		if (asrProperties == null) {
			asrProperties = new Properties();
			FileInputStream propertiesInputStream;
			try {
				propertiesInputStream = new FileInputStream(PROPERTIESFILE);
				asrProperties.load(propertiesInputStream);
				propertiesInputStream.close();
			} catch (Exception e) {System.out.println(e);}
		}
		
		return asrProperties;
	}

	//---------------------------------------------------------------------------------------
	//------------------------------------ Variables ----------------------------------------
	//---------------------------------------------------------------------------------------	
	//******************************** Public Variables *************************************
	
	

	//******************************* Protected Variables ***********************************


	//******************************** Private Variables ************************************

	
}

File: frontend/classes/AdShotRunner/System/ASRProperties.php
<?PHP
/**
* Contains the class for retrieving properties information for the ASR system
*
* @package Adshotrunner
* @subpackage Classes
*/

namespace AdShotRunner\System;

/**
* The ASRProperties class returns system properties for the AdShotRunner system. Such
* properties include database permissions and message queues.
*/
class ASRProperties {

	//---------------------------------------------------------------------------------------
	//---------------------------------- Constants ------------------------------------------
	//---------------------------------------------------------------------------------------	
	const PROPERTIESFILE = RESTRICTEDPATH . "asrProperties.ini";

	//---------------------------------------------------------------------------------------
	//--------------------------------- Static Variables ------------------------------------
	//---------------------------------------------------------------------------------------
	//***************************** Private Static Variables ********************************
	private static $properties = null;

	//---------------------------------------------------------------------------------------
	//---------------------------------- Static Methods -------------------------------------
	//---------------------------------------------------------------------------------------
	//***************************** Public Static Methods ***********************************

	//Database Methods
	public static function asrDatabaseHost() {return (self::getProperties())['asrDatabase']['host'];}
	public static function asrDatabaseUsername() {return (self::getProperties())['asrDatabase']['username'];}
	public static function asrDatabasePassword() {return (self::getProperties())['asrDatabase']['password'];}
	public static function asrDatabase() {return (self::getProperties())['asrDatabase']['database'];}

	//Database Methods
	public static function asrDomain() {return (self::getProperties())['domains']['asrDomain'];}

	//Database Methods
	public static function awsAccessKey() {return (self::getProperties())['aws']['accessKey'];}
	public static function awsSecretKey() {return (self::getProperties())['aws']['secretKey'];}
	public static function awsRegion() {return (self::getProperties())['aws']['region'];}

	//Storage containers
	public static function containerForCampaignJobs() {return (self::getProperties())['storageContainers']['campaignJobs'];}
	public static function containerForPowerPoints() {return (self::getProperties())['storageContainers']['powerPoints'];}
	public static function containerForPowerPointBackgrounds() {return (self::getProperties())['storageContainers']['powerPointBackgrounds'];}
	public static function containerForScreenshots() {return (self::getProperties())['storageContainers']['screenshots'];}
	public static function containerForCreativeImages() {return (self::getProperties())['storageContainers']['creativeimages'];}
	public static function containerForTagPages() {return (self::getProperties())['storageContainers']['tagPages'];}
	public static function containerForTagTexts() {return (self::getProperties())['storageContainers']['tagTexts'];}

	//DFP credentials
	public static function dfpClientID() {return (self::getProperties())['dfp']['clientID'];}
	public static function dfpClientSecret() {return (self::getProperties())['dfp']['clientSecret'];}
	public static function dfpRefreshToken() {return (self::getProperties())['dfp']['refreshToken'];}
	public static function dfpApplicationName() {return (self::getProperties())['dfp']['applicationName'];}

	//Email Addresses
	public static function emailAddressDoNotReply() {return (self::getProperties())['emailAddresses']['doNotReply'];}
	public static function emailAddressContact() {return (self::getProperties())['emailAddresses']['contact'];}
	public static function emailAddressInfo() {return (self::getProperties())['emailAddresses']['info'];}
	public static function emailAddressSupport() {return (self::getProperties())['emailAddresses']['support'];}

	//Message queues
	public static function queueForTagImageRequests() {return (self::getProperties())['messageQueues']['tagImageRequests'];}
	public static function queueForScreenshotRequests() {return (self::getProperties())['messageQueues']['screenshotRequests'];}

	//Notification groups
	public static function notificationGroupForSSSIssues() {return (self::getProperties())['notificationGroups']['sssIssues'];}

	//PowerPoint
	public static function powerPointDefaultBackgroundImage() {return (self::getProperties())['powerPoint']['defaultBackgroundImage'];}
	public static function powerPointDefaultTitle() {return (self::getProperties())['powerPoint']['defaultTitle'];}
	public static function powerPointDefaultFont() {return (self::getProperties())['powerPoint']['defaultFont'];}

	//***************************** Private Static Methods **********************************
	private static function getProperties() {

		//If the properties have not been retrieved, get them from the file
		if (self::$properties == null) {
			self::$properties = parse_ini_file(self::PROPERTIESFILE, true);
		}

		return self::$properties;
	}

	//---------------------------------------------------------------------------------------
	//------------------------------------ Variables ----------------------------------------
	//---------------------------------------------------------------------------------------	
	//******************************** Private Variables ************************************

	//---------------------------------------------------------------------------------------
	//------------------------ Constructors/Copiers/Destructors -----------------------------
	//---------------------------------------------------------------------------------------

	//---------------------------------------------------------------------------------------
	//------------------------------- Modification Methods ----------------------------------
	//---------------------------------------------------------------------------------------
	//********************************* Public Methods **************************************

	//******************************** Protected Methods ************************************

	//********************************* Private Methods *************************************



}


//---------------------------------------------------------------------------------------
//-------------------------- Helper Functions and Wrappers-------------------------------
//---------------------------------------------------------------------------------------

